machine/mellanox/mlnx_x86/rootconf/sysroot-init/mellanox-machine-model.sh

```diff
@@ -1,47 +1,10 @@
 #!/bin/sh
 
-PATH=/usr/bin:/usr/sbin:/bin:/sbin
+cat /etc/machine-build.conf > /etc/machine.conf
```

Comment:
Please do not overwrite `/etc/machine.conf`.  Your `gen-config-platform` should generate everything you need into `/etc/machine-live.conf`.  Then the base `/etc/machine.conf` will take care of the rest.

I think you should just remove this file : `mellanox-machine-model.sh`.  It is not needed anymore with the new common infrastructure.

Reply:
I checked this ONIE with supported NOS's for our systems: Cumulus 3.4.3, Sonic, Mlnx-OS and ONL.
There is a problem with ONL. It looks that ONL installer reads directly /etc/machine.conf
and don't accept new format.
"...
ImportError: No module named "${onie_arch}_${onie_machine}_r${onie_machine_rev}"                              
Install failed. 
"
More than that, it fails also in simply concatenated machine-build.conf and machine-live.conf.
"
...
ValueError: invalid line in /etc/machine.conf: # Runtime ONIE Machine                    
Install failed. 
"
That's why I remove this comment in mellanox-machine-model.sh
Unfortunately we already have systems with ONL. I tested with latest ONL build.
Is it possible that also other NOS rely on old machine.conf format?

Reply:
NOSes should not try to read/parse `/etc/machine.conf` directly.  That file is not an interface.

The supported method is to use `onie-sysinfo` to obtain system information.  See:
https://opencomputeproject.github.io/onie/cli/index.html#onie-sysinfo

If a NOS cannot get what it needs from `onie-sysinfo` then please file a bug and let's fix it.

I understand for the legacy of ONL you may be stuck with this hybrid.  I would encourage ONL to change.